wallet/src/de/schildbach/wallet/service/BlockchainStateDataProvider.kt

```diff
@@ -67,14 +74,79 @@ class BlockchainStateDataProvider @Inject constructor(
         const val MASTERNODE_COUNT = 3800
     }
 
+    private val coroutineScope = CoroutineScope(Dispatchers.IO)
+
     override suspend fun getState(): BlockchainState? {
-        return blockchainStateDao.loadSync()
+        return blockchainStateDao.getState()
     }
 
     override fun observeState(): Flow<BlockchainState?> {
         return blockchainStateDao.observeState().distinctUntilChanged()
     }
 
+    fun updateImpediments(impediments: Set<Impediment>) {
+        coroutineScope.launch {
+            val blockchainState = blockchainStateDao.getState()
+            if (blockchainState != null) {
+                blockchainState.impediments.clear()
+                blockchainState.impediments.addAll(impediments)
+                blockchainStateDao.saveState(blockchainState)
+            }
+        }
+    }
+
```

Comment:
These are all run and forget methods which update the database table row, but they were also run and forget using Executors previously.